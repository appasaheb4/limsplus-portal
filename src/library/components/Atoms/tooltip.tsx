import React, { useRef } from "react"

interface TooltipProps {
  tooltipText?: any
}

const Tooltip: React.FunctionComponent<TooltipProps> = (props) => {
  const tipRef = useRef(null)

  const handleMouseEnter = (tipRef) => {
    tipRef.current.style.opacity = 1
    // tipRef.current.style.marginLeft = "20px"
  }
  function handleMouseLeave(tipRef) {
    tipRef.current.style.opacity = 0
    // tipRef.current.style.marginLeft = "10px"
  }
  return (
    <div
      className="relative flex items-center"
      onMouseEnter={() => handleMouseEnter(tipRef)}
      onMouseLeave={() => handleMouseLeave(tipRef)}
    >
      <div
        className="absolute whitespace-no-wrap bg-gradient-to-r from-black to-gray-700 text-white px-4 py-2 rounded flex items-center transition-all duration-150 z-50"
        style={{ left: -20, bottom: -50, opacity: 0 }}
        ref={tipRef}
      >
        <div
          className="bg-black h-3 w-3 absolute"
          style={{ top: "-6px", transform: "rotate(45deg)" }}
        />
        <div dangerouslySetInnerHTML={{ __html: props.tooltipText }} />
      </div>
      {props.children}
    </div>
  )
}
export default Tooltip
