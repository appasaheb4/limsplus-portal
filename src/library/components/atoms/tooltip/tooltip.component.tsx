import React, {useRef} from 'react';
import * as Material from '@mui/material';

interface TooltipProps {
  tooltipText?: any;
  position?: 'bottom' | 'left' | 'top';
  className?: string;
  children?: any;
}

export const Tooltip: React.FunctionComponent<TooltipProps> = ({
  tooltipText,
  position = 'bottom-start',
  children,
}) => {
  const positionRef = React.useRef<{x: number; y: number}>({
    x: 0,
    y: 0,
  });
  const popperRef = React.useRef<any>(null);
  const areaRef = React.useRef<HTMLDivElement>(null);

  const handleMouseMove = (event: React.MouseEvent) => {
    positionRef.current = {x: event.clientX, y: event.clientY};

    if (popperRef.current != null) {
      popperRef.current.update();
    }
  };
  return (
    <Material.Tooltip
      title={tooltipText}
      placement='bottom-start'
      arrow
      PopperProps={{
        popperRef,
        anchorEl: {
          getBoundingClientRect: () => {
            return new DOMRect(
              positionRef.current.x,
              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
              areaRef.current!.getBoundingClientRect().y + 22,
              0,
              0,
            );
          },
        },
      }}
    >
      <Material.Box ref={areaRef} onMouseMove={handleMouseMove}>
        {children}
      </Material.Box>
    </Material.Tooltip>
  );
};
