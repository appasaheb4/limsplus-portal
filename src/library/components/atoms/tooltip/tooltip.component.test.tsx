import React from 'react';
import {render} from '@testing-library/react';
import {Tooltip} from './tooltip.component';

describe('Tooltip component', () => {
  it('render tooltip correctly', () => {
    const tooltip = render(<Tooltip />);
    expect(tooltip).toMatchSnapshot();
  });
});
