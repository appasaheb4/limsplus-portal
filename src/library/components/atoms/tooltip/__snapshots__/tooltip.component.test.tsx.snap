// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component render tooltip correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="undefined relative flex items-center"
      >
        <div
          class="absolute  whitespace-no-wrap bg-gradient-to-r from-black to-gray-700 text-white px-4 py-2 rounded flex items-center transition-all duration-150 z-50"
          style="left: -20px; bottom: -50px; opacity: 0;"
        >
          <div
            class="bg-black h-3 w-3 absolute"
            style="top: -6px; transform: rotate(45deg);"
          />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="undefined relative flex items-center"
    >
      <div
        class="absolute  whitespace-no-wrap bg-gradient-to-r from-black to-gray-700 text-white px-4 py-2 rounded flex items-center transition-all duration-150 z-50"
        style="left: -20px; bottom: -50px; opacity: 0;"
      >
        <div
          class="bg-black h-3 w-3 absolute"
          style="top: -6px; transform: rotate(45deg);"
        />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
